controllers/datadogagent/feature/test/testsuite.go

```diff
@@ -40,7 +40,7 @@ type FeatureTest struct {
 	DDAv1   *v1alpha1.DatadogAgent
 	Options *Options
 	// Dependencies Store
-	StoreOption        *dependencies.StoreOptions
+	StoreOption        dependencies.StoreOptions
```

Comment:
When I set a custom `StoreOption` in the APM feature tests to store the k8s version for the local service, the v2alpha1 tests with APM enabled would fail with: `testsuite.go:169: feature.ManageDependencies() error = store.AddOrUpdate, no kind is registered for the type v2alpha1.DatadogAgent in scheme "github.com/DataDog/datadog-operator/controllers/datadogagent/feature/test/testsuite.go:95", wantErr false`

From some extra logging, the `StoreOption.Scheme` was populated based on the [first test in the list](https://github.com/DataDog/datadog-operator/blob/49a209fe0aedff11985ea0604f3fdf4a717879b4/controllers/datadogagent/feature/apm/feature_test.go#L38-L43) and that data persisted for the remaining tests instead of being set to `nil` again to be repopulated based on the DDA version. Since the first test is for v1alpha, [testScheme()](https://github.com/DataDog/datadog-operator/blob/3c5f80ad6710ece6800b6287c133668d19b0296b/controllers/datadogagent/feature/test/testsuite.go#L94) used the `v1alpha1.DatadogAgent` type. The scheme wasn't cleared for the v2alpha1 tests so [v2alpha1.DatadogAgent](https://github.com/DataDog/datadog-operator/blob/3c5f80ad6710ece6800b6287c133668d19b0296b/controllers/datadogagent/feature/test/testsuite.go#L112) type wasn't registered.